src/plots/rectanglePlot.ts

```diff
@@ -213,6 +213,45 @@ export module Plots {
       return this;
     }
 
+    /*
+     * Gets the Entities at a particular Point.
+     * 
+     * @param {Point} p
+     * @param {PlotEntity[]} 
+     */
+    public entitiesAt(p: Point) {
+      var results: PlotEntity[] = [];
+      var entities = this.entities();
+      var xScale: Scale<any, number> = this.x().scale;
+      var yScale: Scale<any, number> = this.y().scale;
+      for (var i = 0; i < entities.length; i++) {
+        var entity = entities[i];
+        var x = xScale.scale(this.x().accessor(entity.datum, entity.index, entity.dataset));
+        var y = yScale.scale(this.y().accessor(entity.datum, entity.index, entity.dataset));
+        var x2 = x;
+        if (this.x2()) {
+          x2 = xScale.scale(this.x2().accessor(entity.datum, entity.index, entity.dataset));
+        } else {
+          var xRangeBand = (<Plottable.Scales.Category> xScale).rangeBand();
```

Comment:
This is not coherent. Why is the xScale a category scale if there is no `x2()` accessor? Is it always the case? What if the user just forgot to specify the `x2()`, but uses a linear scale? Is that not a scenario?